From b336eaf001b1950906ae9dc3633a0f48290457aa Mon Sep 17 00:00:00 2001
From: Ben Hutchings <ben@decadent.org.uk>
Date: Fri, 6 Aug 2010 01:35:21 +0100
Subject: [PATCH] CIFS: Include <linux/init.h> in dns_resolve.h

fs/cifs/dns_resolve.c fails to build on some architectures:

In file included from fs/cifs/dns_resolve.c:29:
fs/cifs/dns_resolve.h:27: error: expected '=', ',', ';', 'asm' or '__attribute__' before 'cifs_init_dns_resolver'
fs/cifs/dns_resolve.h:28: error: expected '=', ',', ';', 'asm' or '__attribute__' before 'cifs_exit_dns_resolver'

We need to include <linux/init.h> to ensure that __init and __exit
are defined.

Signed-off-by: Ben Hutchings <ben@decadent.org.uk>
---
 fs/cifs/dns_resolve.h |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/fs/cifs/dns_resolve.h b/fs/cifs/dns_resolve.h
index 26b9eaa..e55fa28 100644
--- a/fs/cifs/dns_resolve.h
+++ b/fs/cifs/dns_resolve.h
@@ -24,6 +24,7 @@
 #define _DNS_RESOLVE_H
 
 #ifdef __KERNEL__
+#include <linux/init.h>
 extern int __init cifs_init_dns_resolver(void);
 extern void __exit cifs_exit_dns_resolver(void);
 extern int dns_resolve_server_name_to_ip(const char *unc, char **ip_addr);
-- 
1.7.1

